pkg/ui/react-app/src/thanos/pages/errorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { ErrorInfo } from 'react';
+import { Container, UncontrolledCollapse, Button } from 'reactstrap';
+import styles from './ErrorBoundary.module.css';
+
+interface ErrorState {
+  error: Error | null;
+  errorInfo: ErrorInfo | null;
+}
+
+class ErrorBoundary extends React.Component<{}, ErrorState> {
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      error: null,
+      errorInfo: null,
+    };
+  }
+
+  componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
+    this.setState({
+      error,
+      errorInfo,
+    });
+  }
+  render(): React.ReactNode {
+    if (this.state.errorInfo) {
+      return (
+        <Container fluid className={styles.container}>
+          <h1>Aaaah! Something went wrong.</h1>
```

Comment:
Just curious, wonder if inline HTML makes sense. What would be the benefit between this and a separate file for HTML (:

Reply:
Sorry, I think I am not able to understand what are you trying to say. This isn't actually inline HTML, it's JSX and it's how we are supposed to work in React. There's only a single `index.html` in the whole React project. :)

Reply:
yea :facepalm: Looks good then (: